lib/modulr/resources/payments/payment.rb

```
@@ -51,18 +52,72 @@ def initialize(raw_response, attributes = {})
           attributes[:details].dig(:details, :fpsTransaction, :paymentInfo, :endToEndId)
         end
 
+        private def parse_attributes(attributes)
+          parse_details(attributes)
+          parse_scheme if type
+        end
+
         private def parse_details(attributes)
           details = attributes[:details]
+          detail_type = details&.dig(:type) || details&.dig(:destinationType)
 
-          case details&.dig(:type) || details&.dig(:destinationType)
+          case detail_type
           when "PI_SECT", "PI_SEPA_INST", "PI_FAST", "PI_REV"
-            Details::Incoming::General.new(nil, details)
+            incoming_detail(details)
           when "ACCOUNT"
-            Details::Incoming::Internal.new(nil, details)
+            incoming_internal_details(details)
+          else
+            outgoing_detail(details)
+          end
+        end
+
+        private def parse_scheme
+          case type
+          when "PI_SECT", "PO_SECT"
+            sepa_regular
+          when "PI_SEPA_INST", "PO_SEPA_INST"
+            sepa_instant
+          when "PI_FAST", "PO_FAST"
+            faster_payments
+          when "ACCOUNT", "INT_INTERC"
+            internal
           else
-            Details::Outgoing::General.new(nil, details)
+            @network = nil
+            @scheme = nil
```

Should we raise an error here instead of null values? 🤔

---

```suggestion
            raise "Unable to find network and scheme for payment with ID: #{id} and Type: #{type}"
```